File: REST_Client.py
import requests
base_url = "http://localhost:5000/user/%s"
while True:
    operacao = input("Escolha uma operação (GET, POST, PUT, DELETE, SAIR): ")
    if operacao.upper() == "GET":
        print(">>> GET <<<")
        id = input("Id: ")
        response = requests.get(base_url % id)
        print(response.json())
        print("HTTP Code: %s" % response.status_code)
    elif operacao.upper() == "POST":
        print(">>> POST <<<")
        id = input("Id: ")
        nome = input("Nome: ")
        idade = input("Idade: ")
        payload = {'nome':nome, 'idade':idade}
        response = requests.post(base_url % id, json=payload)
        print(response.json())
        print("HTTP Code: %s" % response.status_code)
    elif operacao.upper() == "PUT":
        print(">>> PUT <<<")
        id = input("Id: ")
        nome = input("Nome: ")
        idade = input("Idade: ")
        payload = {'nome': nome, 'idade': idade}
        response = requests.put(base_url % id, json=payload)
        print(response.json())
        print("HTTP Code: %s" % response.status_code)
    elif operacao.upper() == "DELETE":
        print(">>> DELETE <<<")
        id = input("Id: ")
        response = requests.delete(base_url % id)
        print(response.json())
        print("HTTP Code: %s" % response.status_code)
    else:
        print(">>> SAIR <<<")
    break

File: REST_Client.js
const url = "http://127.0.0.1:5000/user/";

$('#busca-usuario').submit(function(e){

    e.preventDefault();

   let id = $("#busca-usuario input[name='id']").val();

    $.get(url +  id.trim()).done(function(response){

        console.log(response);
    })
   
})

$('#cadastro-usuario').submit(function(e){

    e.preventDefault();

   let id = $("#cadastro-usuario input[name='id']").val();
   let nome = $("#cadastro-usuario input[name='nome']").val();
   let idade = $("#cadastro-usuario input[name='idade']").val();

    
   $.ajax(url + id.trim(),{
    type: "POST",   
    processData: false,
    data: JSON.stringify({nome:nome, idade:idade}),
    contentType: 'application/json'
    
    
   
  }).done(function(response){

    console.log(response);
  });
   
})

$('#atualiza-usuario').submit(function(e){

    e.preventDefault();

   let id = $("#atualiza-usuario input[name='id']").val();
   let nome = $("#atualiza-usuario input[name='nome']").val();
   let idade = $("#atualiza-usuario input[name='idade']").val();

    
   $.ajax(url + id.trim(),{
    type: "PUT",   
    processData: false,
    data: JSON.stringify({nome:nome, idade:idade}),
    contentType: 'application/json'
    
    
   
  }).done(function(response){

    console.log(response);
  });
   
})

$('#deleta-usuario').submit(function(e){

    e.preventDefault();

   let id = $("#deleta-usuario input[name='id']").val();
   

    
   $.ajax(url + id.trim(),{
    type: "DELETE",   
    processData: false,    
    contentType: 'application/json'
    
    
   
  }).done(function(response){

    console.log(response);
  });
   
})